src/index.ts

```diff
@@ -57,12 +102,12 @@ export default function Memoize(callback, opts = {}) {
 	 * @param {...*} args - Any number of arguments
 	 * @returns {Promise} Value of the callback
 	 */
-	return async (...args) => {
+	return async (...args: any) => {
 		// Serialize the keys...
 		const key = getKey(...args);
```

Comment:
Interestingly, TS flags this with the error `A spread argument must either have a tuple type or be passed to a rest parameter`. And it seems to have a point: If the callback function has multiple parameters, does it make sense to pass those to e.g. `JSON.stringify` as individual params?

Maybe they should be passed like so?

```suggestion
		const key = getKey(args);
```

Reply:
I tried your implementation... this certainly proves a sticking point. I would like to to have the same parameter structure as the callback function. But yes, Typescript is happier if i pass this through as an array of the first parameter

Reply:
Yeah, maybe there's a way to shut up that TS error.